libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -619,7 +590,7 @@ static bool incomingPacketValid( uint8_t packetType )
         case ( MQTT_PACKET_TYPE_PUBREL & 0xF0U ):
 
             /* The second bit of a PUBREL must be set. */
-            if( packetType & 0x02U )
+            if( ( packetType & 0x02U ) > 0U )
```

Comment:
Is this really a MISRA violation? I thought that `if( <value> )` was a common idiom in C.

Reply:
Yes. This was a violation. 

Event misra_c_2012_rule_14_4_violation: | The condition expression "packetType & 2U" does not have an essentially boolean type
-- | --